Packs/PAN-OS/Playbooks/PAN-OS_-_Configure_DNS_Sinkhole.yml

```diff
@@ -0,0 +1,1961 @@
+  playbookInputQuery:
+- key: PublicDNSServerIP
+  value:
+    simple: 8.8.8.8
+  required: true
+  description: |-
+    The IP of the public DNS server that the organization is using to resolve external domains. This is needed to find the rule that allows DNS requests so that malicious ones can be sinkholed by attaching a security profile to that rule.
+
+    If there is currently no rule to allow DNS traffic from the internal DNS server to the internet, the user will be prompted to approve the creation of a new rule that allows DNS traffic from the internal DNS server/s to the IP specified in this input's value.
+  playbookInputQuery:
+- key: RuleNameToAllowDNS
+  value:
+    simple: Internal DNS to External DNS - Allow DNS Traffic
+  required: false
+  description: |-
+    Optional - the name of the security rule that will allow DNS traffic.
+    If permission is granted and no other rule is currently allowing DNS, the playbook will create a new rule with the name specified in this value to allow DNS traffic from inside the network to the outside internet.
+  playbookInputQuery:
+- key: AutoCreateRuleToAllowDNS
+  value:
+    simple: "False"
+  required: false
+  description: |-
+    Determines whether the playbook can automatically create a rule that specifically allows DNS traffic from the internal DNS servers to the public DNS server configured in the playbook inputs.
+    Possible values and their meanings are as follows:
+    True - create such a rule automatically if one does not exist (Warning - this will allow previously unallowed traffic).
+    False - do not create such a rule automatically.
+
+    If a value of "False" is specified, the playbook will check if the NetworkAdminEmail input is specified. If it is, the playbook will seek approval to create the rule, from the email specified in that input. If no email is specified, the playbook will abort without making any changes.
+  playbookInputQuery:
+- key: NetworkAdminEmail
+  value: {}
+  required: false
+  description: |-
+    Optional - the email of the person who should approve the creation of a new security policy rule in the firewall. If no rule allows internal DNS traffic to go outside, the playbook will send an email to the email specified in this input, asking to approve the creation of such rule.
+    If no email is specified, and the AutoCreateRuleToAllowDNS input is set to False, the playbook will abort without making any changes, as such a rule is mandatory for the flow of the playbook.
+  playbookInputQuery:
+- key: RuleNameToDenyToSinkhole
+  value:
+    simple: Deny To Sinkhole
+  required: false
+  description: |-
+    The name of the rule that will be created  to deny traffic to the sinkhole address. This is required so that traffic logs will reveal the endpoints attempting to connect to their C&C (which by the end of this playbook will be sinkholed).
```

Comment:
```suggestion
    The name of the rule that will be created to deny traffic to the sinkhole address. This is required so that traffic logs will reveal the endpoints attempting to connect to their C&C (which by the end of this playbook will be sinkholed).
```